src/QsCompiler/CompilerOptimizations/PureCircuitFinding.fs

```diff
@@ -13,8 +14,18 @@ open PureCircuitAPI
 open OptimizingTransformation
 
 
+/// Returns whether an expression is an operation call
+let private isOperation expr =
```

Comment:
Seems like this would break if we at some point decided to allow that returned values are ignored instead of requiring that they are always assigned. Since this is easy to avoid I suggest to adapt this to not having to rely on this implicit assumption.

Reply:
What do you mean? I don't see what returning values has to do with checking whether an expression is an operation call.

Reply:
Sorry, I was referring to the fact that it does not check whether the given expression is a partial application.

Reply:
Fixed in latest commit